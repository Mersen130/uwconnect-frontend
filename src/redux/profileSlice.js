import { createSlice } from '@reduxjs/toolkit'

export const profileSlice = createSlice({
    name: 'userProfile',
    initialState: {
        email: '',
        userName: '',
        firstName: '',
        lastName: '',
        imgURL: '',
        gender: '',
        faculty: '',
        program: '',
        year: 0,
        courses: [],
        tags: [],
        profileVisible: true,
        agreement: false,
        id: '',
    },
    reducers: {
        setEmail: (state, action) => {
            state.email = action.payload
        },
        setUserName: (state, action) => {
            state.userName = action.payload
        },
        setFirstName: (state, action) => {
            state.firstName = action.payload
        },
        setLastName: (state, action) => {
            state.lastName = action.payload
        },
        setImgURL: (state, action) => {
            state.imgURL = action.payload
        },
        setGender: (state, action) => {
            state.gender = action.payload
        },
        setFaculty: (state, action) => {
            state.faculty = action.payload
        },
        setProgram: (state, action) => {
            state.program = action.payload
        },
        setYear: (state, action) => {
            state.year = action.payload
        },
        setCourses: (state, action) => {
            state.courses = action.payload
        },
        setTags: (state, action) => {
            state.tags = action.payload
        },
        setProfileVisible: (state, action) => {
            state.profileVisible = action.payload
        },
        setAgreement: (state, action) => {
            state.agreement = action.payload
        },
        setID: (state, action) => {
            state.id = action.payload
        },
    },
})

export const { setCourses, setEmail, setUserName, setFaculty, setGender, setID, setImgURL, setProfileVisible, setProgram, setTags, setFirstName, setLastName, setAgreement, setYear } = profileSlice.actions

export default profileSlice.reducer
