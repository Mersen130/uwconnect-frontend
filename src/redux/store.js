import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage'; // defaults to localStorage for web
import profileReducer from './profile/profileSlice';
import authReducer from './auth/authSlice';

const persistConfig = {
  key: 'root',
  storage,
  //whitelist: ['auth'] // persist only the auth slice of the state
};

const rootReducer = combineReducers({
  user: profileReducer,
  auth: authReducer
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer
});

export const persistor = persistStore(store);
