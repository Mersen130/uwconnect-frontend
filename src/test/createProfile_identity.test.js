import './matchMedia.mock';
import GetIdentity from '../route/profile/identity';
import {render,screen,cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import {
    BrowserRouter as Router,
    Routes,
    Route,
    redirect,
    Link
  } from "react-router-dom";
  import { Provider } from 'react-redux'
  import { store } from '../redux/store'

// This is to create ReduxProvider for mock
const ReduxProvider = ({ children, reduxStore }) => (
<Provider store={reduxStore}>{children}</Provider>
)

afterEach(() => {
    cleanup();
})

// This is due to Jest issue of current version
// https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation(query => ({
          matches: false,
          media: query,
          onchange: null,
          addListener: jest.fn(), // deprecated
          removeListener: jest.fn(), // deprecated
          addEventListener: jest.fn(),
          removeEventListener: jest.fn(),
          dispatchEvent: jest.fn(),
        })),
    });
})


test('should render enrollment component', () => {
    render(<ReduxProvider reduxStore={store}><Router><GetIdentity/></Router></ReduxProvider>);
    const loginElement = screen.getByTestId('identity-1');
    // const loginElement = screen.getByPlaceholderText("Faculty");
    // const loginElement = screen.getByText("Faculty");
    expect(loginElement).toBeInTheDocument();
} )


test('should render CreateProfile component', () => {
    render(<ReduxProvider reduxStore={store}><Router><GetIdentity/></Router></ReduxProvider>);
    const loginElement = screen.getByText("--Select Profile Picture--");
    expect(loginElement).toBeInTheDocument();
} )

test('should render CreateProfile component', () => {
    render(<ReduxProvider reduxStore={store}><Router><GetIdentity/></Router></ReduxProvider>);
    const loginElement = screen.getByPlaceholderText("First Name");
    expect(loginElement).toBeInTheDocument();
} )

test('should render CreateProfile component', () => {
    render(<ReduxProvider reduxStore={store}><Router><GetIdentity/></Router></ReduxProvider>);
    const loginElement = screen.getByPlaceholderText("Last Name");
    expect(loginElement).toBeInTheDocument();
} )


test('matches snapshot', () => {
    const tree = renderer.create(<ReduxProvider reduxStore={store}><Router><GetIdentity/></Router></ReduxProvider>).toJSON();
    expect(tree).toMatchSnapshot();

} )
    
