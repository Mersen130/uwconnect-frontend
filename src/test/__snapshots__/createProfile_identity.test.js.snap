// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="identity-1"
>
  <br />
  <div
    className=" css-b62m3t-container"
    onKeyDown={[Function]}
  >
    <span
      className="css-1f43avz-a11yText-A11yText"
      id="react-select-6-live-region"
    />
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      className="css-1f43avz-a11yText-A11yText"
    />
    <div
      className=" css-1ifcq68-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className=" css-1fdsijx-ValueContainer"
      >
        <div
          className=" css-1550uzu-placeholder"
          id="react-select-6-placeholder"
        >
          --Select Profile Picture--
        </div>
        <div
          className=" css-qbdosj-Input"
          data-value=""
        >
          <input
            aria-autocomplete="list"
            aria-describedby="react-select-6-placeholder"
            aria-expanded={false}
            aria-haspopup={true}
            autoCapitalize="none"
            autoComplete="off"
            autoCorrect="off"
            className=""
            disabled={false}
            id="react-select-6-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            role="combobox"
            spellCheck="false"
            style={
              Object {
                "background": 0,
                "border": 0,
                "color": "inherit",
                "font": "inherit",
                "gridArea": "1 / 2",
                "label": "input",
                "margin": 0,
                "minWidth": "2px",
                "opacity": 1,
                "outline": 0,
                "padding": 0,
                "width": "100%",
              }
            }
            tabIndex={0}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className=" css-1i7e1ly-IndicatorsContainer"
      >
        <span
          className=" css-1u9des2-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          className=" css-1xc3v61-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-tj5bde-Svg"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <input
    className="form-control"
    defaultValue=""
    onChange={[Function]}
    placeholder="First Name"
    type="text"
  />
  <input
    className="form-control"
    defaultValue=""
    onChange={[Function]}
    placeholder="Last Name"
    type="text"
  />
  <p
    className="errorMsg"
  />
  <div>
    <input
      autoComplete="off"
      checked={false}
      className="btn-check"
      disabled={false}
      id="toggle-check-female"
      onChange={[Function]}
      type="checkbox"
      value="female"
    />
    <label
      className="mx-1 my-1 btn btn-outline-primary"
      disabled={false}
      htmlFor="toggle-check-female"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      Female
    </label>
    <input
      autoComplete="off"
      checked={false}
      className="btn-check"
      disabled={false}
      id="toggle-check-male"
      onChange={[Function]}
      type="checkbox"
      value="male"
    />
    <label
      className="mx-1 my-1 btn btn-outline-primary"
      disabled={false}
      htmlFor="toggle-check-male"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      Male
    </label>
    <input
      autoComplete="off"
      checked={false}
      className="btn-check"
      disabled={false}
      id="toggle-check-other"
      onChange={[Function]}
      type="checkbox"
      value="other"
    />
    <label
      className="mx-1 my-1 btn btn-outline-primary"
      disabled={false}
      htmlFor="toggle-check-other"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      Other
    </label>
  </div>
</div>
`;
